app/workers/irker_worker.rb

```diff
@@ -0,0 +1,127 @@
+require 'json'
+require 'socket'
+
+# Note: extra configuration possible, in config/gitlab.yml :
+# production:
+#   irker:
+#     server_ip: localhost
+#     server_port: 6659
+#     max_channels: 3
+#     # Next one has no default, this is just an example
+#     default_irc_uri: irc://chat.freenode.net/
+# You can place the 'irker' section just below the already-there 'extra' section
+
+class IrkerWorker
+  include Sidekiq::Worker
+
+  def perform(project_id, chans, colors, push_data)
+    project             = Project.find(project_id)
+    path_with_namespace = project.path_with_namespace
+
+    # Get config parameters
+    begin
+      irker_server = Gitlab.config.irker.server_ip
+      irker_port   = Gitlab.config.irker.server_port
+    rescue Settingslogic::MissingSetting
+      irker_server = 'localhost'
+      irker_port   = 6659
+    end
+
+    logger.info "IrkerWorker: messages destination = #{chans}"
+    @channels = chans
+    begin
+      @socket = TCPSocket.new irker_server, irker_port
+    rescue Errno::ECONNREFUSED => e
+      logger.fatal "Can't connect to Irker daemon: #{e}"
+      return false
+    end
+
+    repo_name     = push_data["repository"]["name"];
+    committer     = push_data["user_name"]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.